homeassistant/components/sensor/bom_weather_current.py

```diff
@@ -0,0 +1,188 @@
+"""
+Support for bom.gov.au current condition weather service.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.bom_weather_current
+"""
+
+import logging
+import requests
+import voluptuous as vol
+import datetime
+
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.helpers.entity import Entity
+import homeassistant.helpers.config_validation as cv
+from homeassistant.util import Throttle
+from homeassistant.const import (
+    CONF_MONITORED_CONDITIONS, TEMP_CELSIUS,
+    STATE_UNKNOWN, CONF_NAME)
+
+_RESOURCE = 'http://www.bom.gov.au/fwo/{}/{}.{}.json'
+_LOGGER = logging.getLogger(__name__)
+
+CONF_ZONE_ID = 'zone_id'
+CONF_WMO_ID = 'wmo_id'
+
+MIN_TIME_BETWEEN_UPDATES = datetime.timedelta(seconds=60)
```

Comment:
The bom data updates at the hour and half hour. It then takes about 4 to 7 minutes for the information to propagate to the accessible api.

The sensor runs every minute until it gets the data. Then using the bom time-stamp of the last data it has received it waits 35 min before hitting the bom api again every minute until it gets new data.

The aim of this is to not hit the bom api for the 35min that there is no new data. If this works as intended there should only be a few api data requests every half hour period.

If i just make the delay 30min then if you start HA at 10min before the hour then you will always receive your data 20min late.